# Introduction

___

Milestones 1

Nama: Dwi Adhi Widigda Kartomihardjo

Batch: FTDS-27

Program ini dibuat untuk melakukan statistik deskriptif dan statistik inferensial untuk meningkatkan tingkat retensi pengguna jasa pengiriman jasa makanan di Bangalore dan meningkatkan faktor-faktor dari jasa pengriminan yang mempengaruhi tingkat retensi pengguna.

___
link Tableau: https://public.tableau.com/app/profile/dwi.adhi.widigda.k./viz/Milestones1DwiAdhiWidigdaK_/Dashboard1



# Problem Statement and Dataset Description


## Later Belakang

Saya sebagai data analyst yang bekerja di perusahaan pengiriman makanan di Bangalore ditugaskan untuk meningkatkan tingkat retensi pengguna jasa pengiriman makanan. Sebagai tambahan saya juga akan mencari faktor-faktor apa saya yang mempengaruhi kemauan user menggunakan jasa

## SMART

Specific: Meningkatkan tingkat retensi pengguna jasa pengiriman makanan dengan cara meningkatkan faktor-faktor yang mempengaruhi kemauan user menggunakan jasa tersebut.

Measurable: Berhasil meningkatkan tingkat retensi user sebesar 70%

Achievable: Mengetahui faktor-faktor yang mempengaruhi user menggunakan jasa pengiriman makanan.

Relevant: Peningkatan tingkat retensi dapat diartikan pengguna lebih bersedia menggunakan jasa yang pada gilirannya meningkatkan pendapatan perusahaan.

Time-bound: Proses analisis dan visualisasi data dilakukan selama 2 minggu hari kerja.

## Problem Statement

Perusahaan mengalami penurunan pengunaan jasa pengiriman makanan. Tim marketing mengestimasikan tingak retensi user sebesar 60%. Selain itu perusahaan kekurangan refensi agar user tetap menggunakan jasa. Saya sebagai data analyst yang berkerja diperusahaan pengiriman makanan di Bangalore ditugasakan meningkatkan retensi user.

## Sumber Dataset

Dataset yang digunkan untuk analisis ini diambil dari Online Food Delivery Preferences-Bangalore Region. Dataset ini terdiri dari 388 baris dan 55 kolom. Tipe data tiap kolom dibagi menjadi 49 String, 3 integer, 1 boolean, dan 2 lain nya. Dataset bisa dibuka melalui link ini https://www.kaggle.com/datasets/benroshan/online-food-delivery-preferencesbangalore-region

## Penjabaran Masalah

Penjabaran masalah dengan metode 5W+1H:

5 penjabaran visualisasi
- Kapan saja customer menggunkan jasa untuk memesan makanan?
- Dimana lokasi penyebaran customer di Bangalore?
- Distribusi umur customer yang akan menggunakan jasa lagi?
- Perbandingan gender customer yang akan menggunakan jasa lagi?
- Pekerjaan apa saja user yang memesan makanan?

1 penjabaran deskriptif
- Umur customer yang akan menggunakan jasa lagi?

1 penjabran statistik inferensial
- Faktor apa saja yang mempengaruhi user menggunakan jasa pengiriman makanan lagi?



# Data Preparation

## Import

In [1]:
import pandas as pd
import numpy as np
import plotly_express as px
from scipy import stats
pd.set_option('display.max_columns', 500)

## Data Loading

In [2]:
df = pd.read_csv('onlinedeliverydata.csv')
df.head()

Age  Gender Marital Status Occupation  Monthly Income  \
0   20  Female         Single    Student       No Income   
1   24  Female         Single    Student  Below Rs.10000   
2   22    Male         Single    Student  Below Rs.10000   
3   22  Female         Single    Student       No Income   
4   22    Male         Single    Student  Below Rs.10000   

  Educational Qualifications  Family size  latitude  longitude  Pin code  \
0              Post Graduate            4   12.9766    77.5993    560001   
1                   Graduate            3   12.9770    77.5773    560009   
2              Post Graduate            3   12.9551    77.6593    560017   
3                   Graduate            6   12.9473    77.5616    560019   
4              Post Graduate            4   12.9850    77.5533    560010   

          Medium (P1)   Medium (P2)   Meal(P1) Meal(P2)  \
0  Food delivery apps   Web browser  Breakfast    Lunch   
1  Food delivery apps   Web browser     Snacks   Dinner   
2  Food delivery apps   Direct call      Lunch   Snacks   
3  Food delivery apps       Walk-in     Snacks   Dinner   
4             Walk-in   Direct call      Lunch   Dinner   

                           Perference(P1)  \
0          Non Veg foods (Lunch / Dinner)   
1          Non Veg foods (Lunch / Dinner)   
2          Non Veg foods (Lunch / Dinner)   
3  Veg foods (Breakfast / Lunch / Dinner)   
4          Non Veg foods (Lunch / Dinner)   

                            Perference(P2) Ease and convenient  \
0                    Bakery items (snacks)             Neutral   
1   Veg foods (Breakfast / Lunch / Dinner)      Strongly agree   
2                  Ice cream / Cool drinks      Strongly agree   
3                    Bakery items (snacks)               Agree   
4   Veg foods (Breakfast / Lunch / Dinner)               Agree   

      Time saving More restaurant choices Easy Payment option  \
0         Neutral                 Neutral             Neutral   
1  Strongly agree          Strongly agree      Strongly agree   
2  Strongly agree          Strongly agree             Neutral   
3           Agree          Strongly agree               Agree   
4           Agree                   Agree               Agree   

  More Offers and Discount Good Food quality Good Tracking system  \
0                  Neutral           Neutral              Neutral   
1           Strongly agree           Neutral                Agree   
2                  Neutral          Disagree              Neutral   
3           Strongly agree             Agree                Agree   
4                    Agree           Neutral              Neutral   

     Self Cooking  Health Concern   Late Delivery    Poor Hygiene  \
0         Neutral         Neutral         Neutral         Neutral   
1  Strongly agree  Strongly agree           Agree  Strongly agree   
2        Disagree         Neutral         Neutral           Agree   
3           Agree  Strongly agree         Neutral           Agree   
4           Agree  Strongly agree  Strongly agree           Agree   

  Bad past experience  Unavailability    Unaffordable Long delivery time  \
0             Neutral         Neutral         Neutral              Agree   
1      Strongly agree  Strongly agree  Strongly agree     Strongly agree   
2               Agree           Agree           Agree              Agree   
3            Disagree        Disagree         Neutral              Agree   
4      Strongly agree           Agree        Disagree     Strongly agree   

  Delay of delivery person getting assigned  \
0                                     Agree   
1                            Strongly agree   
2                                     Agree   
3                                     Agree   
4                            Strongly agree   

  Delay of delivery person picking up food Wrong order delivered  \
0                                    Agree                 Agree   
1                           Strongly agree        Strongly agree 

Cell berikut adalah 5 baris pertama dataset

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 388 entries, 0 to 387
Data columns (total 55 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Age                                        388 non-null    int64  
 1   Gender                                     388 non-null    object 
 2   Marital Status                             388 non-null    object 
 3   Occupation                                 388 non-null    object 
 4   Monthly Income                             388 non-null    object 
 5   Educational Qualifications                 388 non-null    object 
 6   Family size                                388 non-null    int64  
 7   latitude                                   388 non-null    float64
 8   longitude                                  388 non-null    float64
 9   Pin code                                   388 non-null    int64  
 10  Medium (P1)               

Bedasarkan summary dataset, tidak ada kolom dengan missing value selain kolom review.

Penamanam kolom data menggunakan spasi harus diganti. Format penamaan yang digunakan adalah lowercase dan '_' untuke pengganti spasi

# Data Cleaning

In [4]:
df.duplicated().sum()

np.int64(102)

Dataset memiliki 102 dataset duplikat. Data ini perlu di drop

In [5]:
df = df.drop_duplicates()

Duplikat di dataset di hilangkan

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 286 entries, 0 to 369
Data columns (total 55 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Age                                        286 non-null    int64  
 1   Gender                                     286 non-null    object 
 2   Marital Status                             286 non-null    object 
 3   Occupation                                 286 non-null    object 
 4   Monthly Income                             286 non-null    object 
 5   Educational Qualifications                 286 non-null    object 
 6   Family size                                286 non-null    int64  
 7   latitude                                   286 non-null    float64
 8   longitude                                  286 non-null    float64
 9   Pin code                                   286 non-null    int64  
 10  Medium (P1)                    

Setelah tidak ada duplikat, sisa data sebanyak 286

In [7]:
del df['Reviews']

Kolom review di drop karena tidak digunakan untuk analisis.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 286 entries, 0 to 369
Data columns (total 54 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Age                                        286 non-null    int64  
 1   Gender                                     286 non-null    object 
 2   Marital Status                             286 non-null    object 
 3   Occupation                                 286 non-null    object 
 4   Monthly Income                             286 non-null    object 
 5   Educational Qualifications                 286 non-null    object 
 6   Family size                                286 non-null    int64  
 7   latitude                                   286 non-null    float64
 8   longitude                                  286 non-null    float64
 9   Pin code                                   286 non-null    int64  
 10  Medium (P1)                    

In [9]:
df.columns = df.columns.str.replace(' ','_')
df.columns = df.columns.str.lower()

Setiap kolom di dataset yang memiliki spasi diganti dengan '_' dan semua kolom di dataset di lower

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 286 entries, 0 to 369
Data columns (total 54 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   age                                        286 non-null    int64  
 1   gender                                     286 non-null    object 
 2   marital_status                             286 non-null    object 
 3   occupation                                 286 non-null    object 
 4   monthly_income                             286 non-null    object 
 5   educational_qualifications                 286 non-null    object 
 6   family_size                                286 non-null    int64  
 7   latitude                                   286 non-null    float64
 8   longitude                                  286 non-null    float64
 9   pin_code                                   286 non-null    int64  
 10  medium_(p1)                    

In [11]:
df.to_csv('clean_df.csv',index=False)

Dataframe di convert ke dalam csv. Digunakan untuk memperbaiki df jika ada perubahan df yang bersih.

In [12]:
df.to_excel('clean_df.xlsx')

Dataframe di convert ke dalam file excel. Akan digunakan saat pebuatan Dashboard

In [13]:
df = pd.read_csv('clean_df.csv')
df

age  gender marital_status      occupation   monthly_income  \
0     20  Female         Single         Student        No Income   
1     24  Female         Single         Student   Below Rs.10000   
2     22    Male         Single         Student   Below Rs.10000   
3     22  Female         Single         Student        No Income   
4     22    Male         Single         Student   Below Rs.10000   
..   ...     ...            ...             ...              ...   
281   25  Female         Single        Employee  More than 50000   
282   21    Male        Married        Employee  More than 50000   
283   22    Male        Married  Self Employeed   10001 to 25000   
284   31    Male        Married        Employee  More than 50000   
285   30    Male        Married        Employee  More than 50000   

    educational_qualifications  family_size  latitude  longitude  pin_code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
281              Post Graduate            6   13.0140    77.5658    560012   
282                       Ph.D            3   12.9306    77.5434    560085   
283                     School            3   12.9820    77.6256    560008   
284                       Ph.D            3   12.9306    77.5434    560085   
285              Post Graduate            6   12.9369    77.6407    560095   

            medium_(p1)   medium_(p2)   meal(p1) meal(p2)  \
0    Food delivery apps   Web browser  Breakfast    Lunch   
1    Food delivery apps   Web browser     Snacks   Dinner   
2    Food delivery apps   Direct call      Lunch   Snacks   
3    Food delivery apps       Walk-in     Snacks   Dinner   
4               Walk-in   Direct call      Lunch   Dinner   
..                  ...           ...        ...      ...   
281  Food delivery apps   Direct call     Snacks   Dinner   
282  Food delivery apps   Web browser      Lunch   Snacks   
283  Food delivery apps       Walk-in     Dinner   Dinner   
284  Food delivery apps   Web browser      Lunch   Snacks   
285  Food delivery apps       Walk-in      Lunch   Snacks   

                             perference(p1)  \
0            Non Veg foods (Lunch / Dinner)   
1            Non Veg foods (Lunch / Dinner)   
2            Non Veg foods (Lunch / Dinner)   
3    Veg foods (Breakfast / Lunch / Dinner)   
4            Non Veg foods (Lunch / Dinner)   
..                                      ...   
281          Non Veg foods (Lunch / Dinner)   
282          Non Veg foods (Lunch / Dinner)   
283          Non Veg foods (Lunch / Dinner)   
284          Non Veg foods (Lunch / Dinner)   
285          Non Veg foods (Lunch / Dinner)   

                              perference(p2) ease_and_convenient  \
0                      Bakery items (snacks)             Neutral   
1     Veg foods (Breakfast / Lunch / Dinner)      Strongly agree   
2                    Ice cream / Cool drinks      Strongly agree   
3                      Bakery items (snacks)               Agree   
4     Veg foods (Breakfast / Lunch / Dinner)               Agree   
..                                       ...                 ...   
281   Veg foods (Breakfast / Lunch / Dinner)   Strongly disagree   
282   Veg foods (Breakfast / Lunch / Dinner)               Agree   
283   Veg foods (Breakfast / Lunch / Dinner)               Agree   
284   Veg foods (Breakfast / Lunch / Dinner)               Agree   
285   Veg foods (Breakfast / Lunch / Dinner)               Agree   

        time_saving more_restaurant_choices easy_payment_option  \
0           Neutral                 Neutral             Neutral   
1 

# Exploration and Analysis

In [14]:
customer = df[df['output'] == 'Yes']
customer

age  gender marital_status      occupation   monthly_income  \
0     20  Female         Single         Student        No Income   
1     24  Female         Single         Student   Below Rs.10000   
2     22    Male         Single         Student   Below Rs.10000   
3     22  Female         Single         Student        No Income   
4     22    Male         Single         Student   Below Rs.10000   
..   ...     ...            ...             ...              ...   
279   22    Male         Single        Employee   10001 to 25000   
282   21    Male        Married        Employee  More than 50000   
283   22    Male        Married  Self Employeed   10001 to 25000   
284   31    Male        Married        Employee  More than 50000   
285   30    Male        Married        Employee  More than 50000   

    educational_qualifications  family_size  latitude  longitude  pin_code  \
0                Post Graduate            4   12.9766    77.5993    560001   
1                     Graduate            3   12.9770    77.5773    560009   
2                Post Graduate            3   12.9551    77.6593    560017   
3                     Graduate            6   12.9473    77.5616    560019   
4                Post Graduate            4   12.9850    77.5533    560010   
..                         ...          ...       ...        ...       ...   
279                   Graduate            1   13.0138    77.5877    560006   
282                       Ph.D            3   12.9306    77.5434    560085   
283                     School            3   12.9820    77.6256    560008   
284                       Ph.D            3   12.9306    77.5434    560085   
285              Post Graduate            6   12.9369    77.6407    560095   

            medium_(p1)   medium_(p2)   meal(p1) meal(p2)  \
0    Food delivery apps   Web browser  Breakfast    Lunch   
1    Food delivery apps   Web browser     Snacks   Dinner   
2    Food delivery apps   Direct call      Lunch   Snacks   
3    Food delivery apps       Walk-in     Snacks   Dinner   
4               Walk-in   Direct call      Lunch   Dinner   
..                  ...           ...        ...      ...   
279  Food delivery apps   Direct call      Lunch   Dinner   
282  Food delivery apps   Web browser      Lunch   Snacks   
283  Food delivery apps       Walk-in     Dinner   Dinner   
284  Food delivery apps   Web browser      Lunch   Snacks   
285  Food delivery apps       Walk-in      Lunch   Snacks   

                             perference(p1)  \
0            Non Veg foods (Lunch / Dinner)   
1            Non Veg foods (Lunch / Dinner)   
2            Non Veg foods (Lunch / Dinner)   
3    Veg foods (Breakfast / Lunch / Dinner)   
4            Non Veg foods (Lunch / Dinner)   
..                                      ...   
279  Veg foods (Breakfast / Lunch / Dinner)   
282          Non Veg foods (Lunch / Dinner)   
283          Non Veg foods (Lunch / Dinner)   
284          Non Veg foods (Lunch / Dinner)   
285          Non Veg foods (Lunch / Dinner)   

                              perference(p2) ease_and_convenient  \
0                      Bakery items (snacks)             Neutral   
1     Veg foods (Breakfast / Lunch / Dinner)      Strongly agree   
2                    Ice cream / Cool drinks      Strongly agree   
3                      Bakery items (snacks)               Agree   
4     Veg foods (Breakfast / Lunch / Dinner)               Agree   
..                                       ...                 ...   
279                                   Sweets               Agree   
282   Veg foods (Breakfast / Lunch / Dinner)               Agree   
283   Veg foods (Breakfast / Lunch / Dinner)               Agree   
284   Veg foods (Breakfast / Lunch / Dinner)               Agree   
285   Veg foods (Breakfast / Lunch / Dinner)               Agree   

        time_saving more_restaurant_choices easy_payment_option  \
0           Neutral                 Neutral             Neutral   
1 

Dataframe customer dibuat dengan filter dari df dimana nilai di kolom output nya 'Yes'

## Penjabaran Visualisasi
- Kapan saja customer menggunkan jasa untuk memesan makanan?
- Dimana lokasi penyebaran customer di Bangalore?
- Distribusi umur customer yang akan menggunakan jasa lagi?
- Perbandingan gender customer yang akan menggunakan jasa lagi?
- Pekerjaan apa saja user yang memesan makanan?

In [15]:
gender_vis = customer.groupby('gender')[['output']].count().reset_index()
fig = px.bar(gender_vis,x='gender',y='output',title='Banyak nya customer bedasarkan gender')
fig.show()

Bedasarkan bar graph
- Pengguna wanita sebanyak 92.
- Pengguna laki-laki sebanyak 129.
- Pengguna laki-laki lebih banyak dari wanita.

In [16]:
meals_vis = customer.groupby('meal(p1)')[['output']].count().reset_index()
fig = px.pie(meals_vis,values='output',names='meal(p1)',title='Proporsi makanan yang biasa dipesan')
fig.show()

Bedasarkan pie chart:
- Proporsi sarapan siang sebanyak 38 atau 17.2%
- Proporsi makan siang sebanyak 71 atau 32.1%
- Proporsi makan malam sebanyak 44 atau 19.9%
- Proporsi snack sebanyak 68 atau 30.8%
- Customer lebih sering menggunakan aplikasi di jam makan siang
- Customer jarang menggunakan aplikasi saat sarapan
- Proporsi customer menggunakan aplikasi untuk snack cukup besar, 
  lebih banyak dari sarapan dan makan malan dan tidak jauh dari makan siang

In [17]:
occupation_vis = customer.groupby('occupation')[['output']].count().reset_index()
fig = px.pie(occupation_vis,values='output',names='occupation',title='Proporsi pekerjaan customer')
fig.show()

Bedasarkan piechart ini:
- Pengguna terbanyak aplikasi adalah mahasiswa.
- Pengguna kedua terbanyak aplikasi adalah karyawan.
- Pengguna terkecil aplikasi adalah ibu rumah tangga.

In [18]:
fig = px.histogram(customer, x='age', title='Distribusi umur pengguna aplikasi')
fig.show()

Bedasarkan histogram:
- Umur pengguna aplikasi terbanyak adalah 23 tahun
- Terdapat 1 user yang berumur 18 tahun
- Rata-rata persebaran pengguna yang menggunakan aplikasi diantara 22 dan 25 tahun. 

## Penjabaran Deskriptif
- Umur user yang akan menggunakan jasa lagi?

In [19]:
mean_umur = customer['age'].mean()
print(f'Rata-rata umur: {mean_umur}')
median_umur = customer['age'].median()
print(f'Nilai tengah umur: {median_umur}')
mode_umur =  customer['age'].mode()[0]
print(f'Umur paling banyak: {mode_umur}')

Rata-rata umur: 24.28054298642534
Nilai tengah umur: 24.0
Umur paling banyak: 23


Rata-rata dan nilai tengah umur dekat yaitu 24.28 dan 24.
Modus umur juga mendekati rata-rata dan nilai tengah yaitu 23

In [20]:
std_umur = customer['age'].std()
print(f'Standar devisai umur: {std_umur}')
variance_umur =customer['age'].var()
print(f'Variance umur: {variance_umur}')

Standar devisai umur: 2.942329268115213
Variance umur: 8.657301522007405


Nilai standard deviasi untuk umur adalah 2.94. Bisa dibilang distribusi nya mendekati nilai rata-rata

In [21]:
skew_umur = customer['age'].skew()
print(f'Skewness umur: {skew_umur}')
kurt_umur = customer['age'].kurtosis()
print(f'Kurtosis umur: {kurt_umur}')

Skewness umur: 1.0419364631370496
Kurtosis umur: 0.9150051752570034


Nilai skewness umur adalah 1.04 dan lebih dari 1 yang artinya distribusinya highly skew
Nilai kurtosis umur adalah 0.91 lebih dari 0 yang aritnya ada outlier dan nilai outlier nya extreme

Tetapi bedasarkan nilai rata-rata dan nilai tengah karena kedua nilai nya dekat bisa dibulang distribusi data umur normal. 
Selain itu bedasarkan histogram umur distribusi data umur relatif normal meskipun ada tail di bagian kanan graph.

# Penjabaran Statistik Inferensial
- Faktor apa saja yang mempengaruhi user menggunakan jasa pengiriman makanan lagi?

In [22]:
columns = df.drop(columns=['age','family_size','latitude','longitude','pin_code','output']).columns
column_dependant = []
for column in columns:
    cross = pd.crosstab(df[column],df['output'])
    pval = stats.chi2_contingency(cross).pvalue
    if pval < 0.05:
        prompt = "kolom ini dependan terhadap ouput"
        column_dependant.append(column)
    else:
        prompt = ""
    print(f'P-value dari kolom {column} and Output: {pval} {prompt}')

P-value dari kolom gender and Output: 0.6130318114453361 
P-value dari kolom marital_status and Output: 7.703401912918085e-06 kolom ini dependan terhadap ouput
P-value dari kolom occupation and Output: 5.67080932624632e-05 kolom ini dependan terhadap ouput
P-value dari kolom monthly_income and Output: 0.004034489801786449 kolom ini dependan terhadap ouput
P-value dari kolom educational_qualifications and Output: 0.11820271758058079 
P-value dari kolom medium_(p1) and Output: 0.07158594379038265 
P-value dari kolom medium_(p2) and Output: 0.9138678754738945 
P-value dari kolom meal(p1) and Output: 0.7009573905875501 
P-value dari kolom meal(p2) and Output: 0.7931282059689989 
P-value dari kolom perference(p1) and Output: 0.42684793780409414 
P-value dari kolom perference(p2) and Output: 0.04244114260516224 kolom ini dependan terhadap ouput
P-value dari kolom ease_and_convenient and Output: 5.923836793706606e-26 kolom ini dependan terhadap ouput
P-value dari kolom time_saving and Output:

Untuk menjawab faktor apa saya yang mempengaruhi user menggunakan jasa pengiriman makanan lagi dapat menggunkana chi square untuk mencari korelasi antara 2 data kategorikal.

H0 untuk chi square adalah tidak ada korelasi antara variabel
h1 untuk chi squrea adalah ada korelasi antara varibel

Kondisi dari chi square ini adalah jika p-value variable > 0.05 maka H0 gagal di tolak dan jika p value < 0.05 maka H0 di tolak.

Kolom numerik seperti 'age','family_size','latitude','longitude','pin_code','output' di potong terlebih dahulu.

In [23]:
column_dependant

['marital_status',
 'occupation',
 'monthly_income',
 'perference(p2)',
 'ease_and_convenient',
 'time_saving',
 'more_restaurant_choices',
 'easy_payment_option',
 'more_offers_and_discount',
 'good_food_quality',
 'good_tracking_system',
 'self_cooking',
 'health_concern',
 'late_delivery',
 'bad_past_experience',
 'unavailability',
 'unaffordable',
 'delay_of_delivery_person_picking_up_food',
 'maximum_wait_time',
 'less_delivery_time',
 'number_of_calls',
 'politeness']

Cell berikut menunjukan kolom dengan nilai p-value < 0.05 
Karena p-value lebih kecil dari 0.05 maka H0 ditolak dan kolom ini mendukung H1.
Kolom-kolom ini memiliki hubungan dengan kolom output.

# Conclusion

## Kesimpulan

Kesimpulan yang dibentuk dari analisis ini adalah user yang akan menggunkan aplikasi nya lagi atau customer cenderung memesan untuk makan siang dan untuk snacks. Umur customer secara rata-rata 24 tahun dengan standar deviasi 2.94 dengan kata lain orang dewasa muda. Selain itu lebih banyak customer laki-laki daripada wanita.



## Rekomendasi

Rekomendasi dari hasil analisis ini adalah tim marketing bisa mempromosi aplikasi merunjuk ke laki-laki berumur 21-27 tahun untuk kalangan mahasiswa dan karyawan. Promosi bisa berupa diskon yang dapat digunakan untuk makan siang atau makanan dengan kategori cemilan. Selain promosi, aplikasi pengiriman dapat ditingkatkan dari segi kemudahan memesan online, opsi restoran yang lebih banyak, kemudahan membayar, memberpanyak diskon, sistem tracking aplikasi, cepat nya pengiriman, sering nya pengantar makanan menelepon customer dan kesopanan personal pengirim makanan. 
Selain itu aplikasi harus menghindari dan meningkatkan faktor-faktor seperti pengiriman yang telat, tidak tersedia nya jasa ,ongkos yang tidak terjangkau, waktu pengiriman yang lama, waktu maksimum customer menunggu. 